.github/workflows/build_directory_md.yml

```diff
@@ -0,0 +1,37 @@
+---
+name: build_directory_md
+
+# yamllint disable-line rule:truthy
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'main'
+    paths-ignore:
+      - '.github/**'
+      - '.scripts/**'
+      - 'DIRECTORY.md'
+
+jobs:
+  build_directory_md:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: jiro4989/setup-nim-action@v1
+
+      - name: Build file
+        run: |
+          git clean -f -x -d
+          nim r .scripts/directory.nim > DIRECTORY.md
+
+      - name: Commit changes
+        run: |
+          date > generated.txt
+          git config user.name github-actions
+          git config user.email github-actions@users.noreply.github.com
+          git add DIRECTORY.md
+          git commit -m "Update DIRECTORY.md"
+          gh pr create --title "Update DIRECTORY.md" --review "@owner" --base "GITHUB_REF/main"
```

Comment:
```suggestion
      - name: Setup Git configurations
        run: |
            git config --global user.name github-actions[bot]
            git config --global user.email 'github-actions@users.noreply.github.com'
      - name: Commit and push changes
        run: |
          git checkout -b directory_update-${{ github.sha }}

          date > generated.txt
          git add DIRECTORY.md

          git commit -m "docs: update DIRECTORY.md"
          git push origin directory_update-${{ github.sha }}:directory_update-${{ github.sha }}
      - name: Create a pull request
        run: |
          if [[ $(git log --branches --not --remotes) ]]; then
              gh pr create --base ${GITHUB_REF##*/} --head directory_update-${{ github.sha }} --title 'docs: update DIRECTORY.md' --body 'Updated the DIRECTORY.md file (see the diff. for changes).'
          fi
```

Reply:
Wouldn't `--local` be better than `--global`?
I can't remember. What does this instruction `date > generated.txt` do?

Reply:
I think `--global` is fine. It would run for the action only.
Not sure either. I just included it from the previous/current script.

Reply:
I removed the `date` command. We do not use it afterwise.